### PR TITLE
Update change log for upcoming release

### PR DESCRIPTION
**Pull request by trask**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,9 +2,49 @@
 
 ## Unreleased
 
+### Migration notes
+
+- Spring Boot Starter Scheduling instrumentation scope name changed from
+  `io.opentelemetry.spring-scheduling-3.1` to `io.opentelemetry.spring-boot-autoconfigure`
+  to reflect the module's name.
+- Default flush timeout for aws lambda flush javaagent instrumentation changed from 1 second
```
Comment (laurit) [inline CHANGELOG.md:10]:
```suggestion
- Default flush timeout for aws lambda javaagent instrumentation changed from 1 second
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,9 +2,49 @@
...
+- Add logback mdc instrumentation to spring boot starter
+  ([#12515](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/12515))
+- Run class load listener only once
+  ([#12565](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/12565))
+- Remove duplicate byte buddy classes to reduce agent jar file size
+  ([#12571](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/12571))
+- Support additional JVM arg syntax in service name resource detector
+  ([#12544](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/12544))
+- Fix warning about setting VirtualField supplier twice
+  ([#12584](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/12584))
```
Comment (laurit) [inline CHANGELOG.md:34]: this was introduced with #12444 so I think we don't need to mention it in the change log

**Review by laurit — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,9 +2,49 @@
...
+
+- Fix derby directory connection string parser
+  ([#12479](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/12479))
+- Improve whitespace handling in oracle jdbc url parser
+  ([#12512](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/12512))
+- Fix SpanKey bridging for unbridgeable span
+  ([#12511](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/12511))
+- Fix lettuce instrumentation and tests to pass against latest version
+  ([#12552](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/12552))
+- fix: Kafka initialization occasionally failed due to concurrent injection of OpenTelemetryMetricsReporter (to #12538)
```
Comment (steverao) [inline CHANGELOG.md:46]:
```suggestion
- Fix Kafka initialization occasionally failed due to concurrent injection of OpenTelemetryMetricsReporter
```
How about keeping same writing style here?

**Review by steverao — COMMENTED** [review]
BTW, it might need to contain https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/12562 now.

**Pull request merged**